doc/src/modules/physics/mechanics/kane.rst

```diff
@@ -22,12 +22,37 @@ differentiated non-holonomic equations.
   \mathbf{k_d}(q, t) \dot{u} + \mathbf{f_d}(q, \dot{q}, u, t) &= 0\\
   \mathbf{k_{dnh}}(q, t) \dot{u} + \mathbf{f_{dnh}}(q, \dot{q}, u, t) &= 0\\
 
-In :mod:`mechanics` holonomic constraints are only used for the linearization
-process; it is assumed that they will be too complicated to solve for the
-dependent coordinate(s).  If you are able to easily solve a holonomic
-constraint, you should consider redefining your problem in terms of a smaller
-set of coordinates. Alternatively, the time-differentiated holonomic
-constraints can be supplied.
+The first equation is non-linear in the generalized coordinates and typically
+describes configuration constraints like kinematic loops. For the purposes of
+forming the non-linear equations of motion, these are additional equations that
+must be satisfied along with the other equations. In :mod:`mechanics` holonomic
+constraints are only used for the linearization process; it is assumed that
+they will be too complicated to solve for the dependent coordinate(s). If you
+are able to easily solve a holonomic constraint, you should consider redefining
+your problem in terms of a smaller set of coordinates. Alternatively, the
+time-differentiated holonomic constraints can be supplied.
+
+The second equation describes the non-holonomic constraints, i.e. velocity
+constraints, that are linear in the generalized speeds. There are fewer
+equations than generalized speeds, and thus describe the relationship between
+the dependent and independent generalized speeds. Where there are :math:`m`
```

Comment:
They are the same. The number of dependent speeds is equal to the number of constraints. For example, if you have 5 generalized speeds and 3 motion constraint equations that are linear in all 5 generalized speeds, then you can solve the 3 equations for three unknowns, thus 3 dependent speeds, leaving 2 independent speeds. I'll clarify.